### ISSUE TITLE
Suppress compiler warning when compiling common/url.c - url_decode with MSVC

### ISSUE THREAD

**Issue by vjardin** [p11-glue/p11-kit#248]
I am not fully familiar yet with all the design of p11-kit, but this code looks strange:
https://github.com/p11-glue/p11-kit/blob/17b2edb1354c5d5080172af45ce55408c5e51d59/common/url.c#L85

It is highlighted by Microsoft `cl.exe` compiler:

> p11-kit\\common\url.c(85): warning C4244: '=' : conversion de '__int64' en 'unsigned char', perte possible de données

(which means conversiion from __int64 to unsigned char, possible lost of information

the compiler is right because:
`const static char HEX_CHARS_UPPER[] = "0123456789ABCDEF";`

Please, what's the idea or purpose of this code ? I see there are some unit tests, but it does not cover this case.

**Comment by vjardin** [comment]
You are right. I get the point. I think I have an idea to avoid this compiler warning. I'll provide a patch soon.

**Comment by vjardin** [comment]
I'll send a fix after the pull request #251

**Comment by winner658** [comment]
这个issue完成修复了么，修复在哪里了
